Add warning when checkpoint_timeout exceeds db-timeout.

pgstef opened the pull request:
In the backup command, add a warning if start-fast is disabled and PG checkpoint_timeout is greater than db-timeout.
In such cases, we might timeout before the checkpoint occurs and the backup really starts.

Tests does seem to run fine now:

```bash
2021-11-10 10:15:59.309 P00   INFO: P1-T1/1 - vm=f33, module=db, test=db (11.63s)
2021-11-10 10:15:59.790 P00   INFO: tested modules have full coverage
2021-11-10 10:15:59.790 P00   INFO: writing C coverage report
2021-11-10 10:15:59.901 P00   INFO: TESTS COMPLETED SUCCESSFULLY (26s)

2021-11-10 10:16:53.501 P00   INFO: P1-T1/1 - vm=f33, module=command, test=backup (33.39s)
2021-11-10 10:16:53.981 P00   INFO: tested modules have full coverage
2021-11-10 10:16:53.981 P00   INFO: writing C coverage report
2021-11-10 10:16:54.196 P00   INFO: TESTS COMPLETED SUCCESSFULLY (35s)
```

pgstef commented on test/src/common/harnessPq.h:
```diff
@@ -68,24 +68,27 @@ Macros for defining groups of functions that implement various queries and comma
         : archiveMode},                                                                                                            \
     {.session = sessionParam, .function = HRNPQ_GETVALUE, .param = "[0,3]", .resultZ = archiveCommand == NULL ? PROJECT_BIN        \
         : archiveCommand},                                                                                                         \
+    {.session = sessionParam, .function = HRNPQ_GETVALUE, .param = "[0,4]", .resultZ = STRINGIFY(versionParam)},                   \
```
Oops, wrong copy/paste. Setting it to 300 also solves the tests being too noisy

pgstef replied:
Solved in 4e88d16 :-)

pgstef commented on test/src/module/command/backupTest.c:
```diff
@@ -2457,6 +2467,8 @@ testRun(void)
             hrnCfgArgRawBool(argList, cfgOptRepoHardlink, true);
+            // With start-fast being disabled, increase db-timeout to be greater than checkpoint_timeout
+            hrnCfgArgRawZ(argList, cfgOptDbTimeout, "110001");
             HRN_CFG_LOAD(cfgCmdBackup, argList);
```
Solved in 4e88d16 :-)

pgstef commented on src/db/db.h:
```diff
@@ -30,6 +30,7 @@ typedef struct DbPub
     const String *archiveCommand;                                   // The archive_command reported by the database
     const String *pgDataPath;                                       // Data directory reported by the database
+    unsigned int checkpointTimeout;                                 // The checkpoint timeout reported by the database
     unsigned int pgVersion;                                         // Version as reported by the database
```
Solved in 4e88d16 :-)

pgstef commented on src/db/db.c:
```diff
@@ -244,11 +245,13 @@ dbOpen(Db *this)
             this->pub.pgDataPath = strDup(varStr(varLstGet(row, 1)));
             this->pub.archiveMode = strDup(varStr(varLstGet(row, 2)));
             this->pub.archiveCommand = strDup(varStr(varLstGet(row, 3)));
+            this->pub.checkpointTimeout = varUIntForce(varLstGet(row, 4));
```
Solved in 4e88d16 :-)

pgstef commented on src/command/backup/backup.c:
```diff
@@ -850,6 +850,17 @@ backupStart(BackupData *backupData)
+                    " reported by the database - timeout may occur before the backup actually starts",
+                    strZ(cfgOptionDisplay(cfgOptDbTimeout)), dbCheckpointTimeout(backupData->dbPrimary));
+            }
+
```
Hm I see, `db-timeout` might be different on repo/db hosts. I'll investigate how to handle the info from `dbOpenProtocol` then :-)

pgstef replied:
The "not remote" part seems to be working now (since 80358f3) but I'm still struggling with `dbOpenProtocol()`.

Should I change the `pckWriteU32P` to push both index and timeout or should I rather change the function return with something like `FUNCTION_LOG_RETURN(UINT64, cfgOptionUInt64(cfgOptDbTimeout));`?

dwsteele replied:
I remembered that we already have an API for pulling options from a remote host, so I added that in 31e71c80d2f1. 

Also, I changed `pgClientTimeout()` to `cfgOptionUInt64(cfgOptDbTimeout)` for the local case. The client timeout is protocol-timeout which is >= db-timeout + 30.

Now we'll see if this passes CI.

pgstef replied:
Oh, that's great, thanks :-)

dwsteele commented:
@pgstef I rebased this at 4d0e91907b6f, which included simplifying the test since there were merge conflicts in that area. I also moved and updated the message at fa17b9cc79bf5. The message needed to be *before* backup start so the user would see the warning in case of a timeout.

This looks ready to me. Is there anything you were waiting for?

pgstef commented:
> @pgstef I rebased this at [4d0e919](https://github.com/pgbackrest/pgbackrest/commit/4d0e91907b6fa6a9ef8d37a76b0ee6a9ecc6c736), which included simplifying the test since there were merge conflicts in that area. I also moved and updated the message at [fa17b9c](https://github.com/pgbackrest/pgbackrest/commit/fa17b9cc79bf52db02ae023793e97bc962934b28). The message needed to be _before_ backup start so the user would see the warning in case of a timeout.

:+1: 

> This looks ready to me. Is there anything you were waiting for?

This was ready for your second round of review. Looks ready to me too, Thanks !
